pages/profile.tsx

```diff
@@ -193,13 +201,21 @@ export default createPage({
               }}
             />
             <label className="form-check-label" htmlFor="flexCheckChecked">
-              Allow others to see my profile&nbsp;
+              Allow others to see my profile
             </label>
             {profile.updatingIsPublic ? (
               <Spinner animation="border" className="mx-auto" size="sm" />
             ) : null}
           </div>
         )}
+        <div className="mt-2">
+          <Button
+            variant="primary"
+            onClick={() => router.push(`/publicprofile?id=${uid}`)}
+          >
+            Preview Profile
```

Comment:
Consider calling this "View Profile" since we're just linking to the profile and there is no save the user has to do to commit the changes.